pubsub-protocol-binding.md

```diff
@@ -0,0 +1,263 @@
+particular content mode might be defined by an application, but are not defined
+here.
+
+The receiver of the event can distinguish between the two content modes by
+inspecting the [Pub/Sub Attribute `Content-Type`][content-type] of the
+[Pub/Sub message][pubsub-message]. If the attribute is present and its value is prefixed 
+with the CloudEvents media type `application/cloudevents`, indicating the use of a known
+[event format](#14-event-formats), the receiver SHOULD use _structured_ mode,
+otherwise it MUST default to _binary_ mode.
+
+If a receiver finds a CloudEvents media type as per the above rule, but with an
+event format that it cannot handle, for instance
+`application/cloudevents+avro`, it MAY still treat the event as binary and
+forward it to another party as-is.
+
+### 3.1. Binary Content Mode
+
+The _binary_ content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. Content-Type
+
+For the _binary_ mode, the [Pub/Sub Attribute `Content-Type`][content-type], if present, MUST 
+be used as the CloudEvents `datacontenttype` attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data`](#22-data-attribute) byte-sequence MUST be used as the value of the Pub/Sub
+message data payload.
+
+#### 3.1.3. Metadata Attributes
+
+All [CloudEvents][ce] attributes 
+MUST be individually mapped to and from distinct Pub/Sub message attributes.
+
+##### 3.1.3.1 Pub/Sub Attribute Names
+
+Except for attributes
+[explicitly handled in this specification](#2-use-of-cloudevents-attributes),
+the naming convention for the Pub/Sub attribute mapping of well-known CloudEvents
+attributes is that each attribute name MUST be prefixed with "ce-".
+
+Examples:
+
+    * `time` maps to `ce-time`
+    * `id` maps to `ce-id`
+    * `specversion` maps to `ce-specversion`
+
+Attribute names are case-insensitive.
+
+##### 3.1.3.2 Pub/Sub Attribute Values
+
+The value for each Pub/Sub attribute MUST be constructed from the respective
+CloudEvents attribute type's [canonical string representation][ce-types].
+
+#### 3.1.4 Examples
+
+This example shows the _binary_ mode mapping of an event to a Pub/Sub Message:
+
+```text
+Publish /some-topic
+--- Attributes ---
+{
+  "ce-specversion": "1.0",
+  "ce-type": "com.example.someevent",
+  "ce-time": "2020-03-10T03:56:24Z",
+  "ce-id": "1234-1234-1234",
+  "ce-source": "/mycontext/subcontext",
+    .... further attributes ...
+  "ce-datacontenttype": "application/json; charset=utf-8",
+}
+--- Data ---
+{
+    ... application data ...
+}
+```
+
+### 3.2. Structured Content Mode
+
+The _structured_ content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple transports.
+
+#### 3.2.1. Content-Type
+
+The [Pub/Sub Attribute `Content-Type`][content-type] MUST be set to the media type of
```

Comment:
Is the casing intentionally?

I'd find `content-type` more consistent with the binary mode. `contentType` is more consistent with the other PubSub attributes (e.g. `messageId`).